rl_bakery/agent_trainer/agents.py

```diff
@@ -0,0 +1,173 @@
+import abc
+import tensorflow as tf
+from tf_agents.utils import common
+from tf_agents.networks import q_network
+from tf_agents.agents.dqn import dqn_agent
+from tf_agents.agents import ddpg
+from tf_agents.trajectories import time_step as ts
+import logging
+
+from rl_bakery.agent_trainer.config import Optimizer
+
+logger = logging.getLogger(__name__)
+
+
+class AgentTrainer(metaclass=abc.ABCMeta):
```

Comment:
Extracting the Agent initialization from RL_Application and placing it on its own is a good idea. 

Here are some suggestions in relation to improving this code:
* Rename the AgentTrainer to AgentInitializerInterface. This interface needs to be implemented to initial the different agents (e.g. dqn, sac...)

* Remove the _init_ function and make let the implementation of this interface handle that. Different types of agents might need different parameters